Examples/Io/Performance/ActsExamples/Io/Performance/VertexPerformanceWriter.cpp

```diff
@@ -511,19 +465,10 @@ ActsExamples::ProcessCode ActsExamples::VertexPerformanceWriter::writeT(
 
       auto& innerTrkWeight = m_trkWeight.emplace_back();
 
-      for (std::size_t j = 0; j < associatedTruthParticles.size(); ++j) {
-        const auto& particle = associatedTruthParticles[j];
-        SimBarcode vtxId =
-            particle.particleId().setParticle(0).setSubParticle(0);
-
-        if (vtxId == maxOccurrenceId) {
-          // Vertex found, fill variables
-
-          // Helper function for computing the pull
-          auto pull = [&](const Acts::ActsScalar& diff,
-                          const Acts::ActsScalar& variance,
-                          const std::string& variableStr,
-                          const bool& afterFit = true) {
+      // Helper function for computing the pull
+      auto pull =
```

Comment:
This lambda is defined inside the vertex loop. Might be worth considering pulling it out and not capturing the loop variables.

Reply:
capturing only `this` now